### PR TITLE
25 support tileid filter with wildcard on cdse

### PR DESCRIPTION
**Pull request by bossie**
Implements #25.

### CODE REVIEW

File: src/test/scala/org/openeo/CreodiasAPITest.scala [hunk 1]
```diff
@@ -42,4 +42,16 @@ class CreodiasAPITest {
     )
   }
 
+  @Test
+  def testTileIdWithWildcardIsNotPropagatedToApi(): Unit = {
+    // propagating it makes the API return a 400 error
+    new CreodiasClient().getProducts(
```
Comment (EmileSonneveld) [inline src/test/scala/org/openeo/CreodiasAPITest.scala:48]: No check if product is indeed omitted?

Comment (bossie) [reply src/test/scala/org/openeo/CreodiasAPITest.scala:48]: Quite right, let me have a look.

File: src/main/scala/org/openeo/opensearch/OpenSearchResponses.scala [hunk 2]
```diff
@@ -692,6 +692,20 @@ object OpenSearchResponses {
...
+
+    private def retainTileIdPattern(features: Array[Feature], tileIdPattern: Option[String]): Array[Feature] =
+      tileIdPattern match {
+        case Some(pattern) => features.filter(feature => feature.tileID match {
+          case Some(tileId) =>
+            val matchesPattern = tileId matches pattern.replace("*", ".*")
+            logger.debug(s"${if (matchesPattern) "retaining" else "omitting"} feature with tileId $tileId")
+            matchesPattern
+          case _ =>
+            logger.debug(s"omitting feature with unknown tileId")
```
Comment (EmileSonneveld) [inline src/main/scala/org/openeo/opensearch/OpenSearchResponses.scala:704]: Ok to add more info here? `logger.info(s"Omitting feature with unknown tileId: " + feature.id)`

Comment (bossie) [reply src/main/scala/org/openeo/opensearch/OpenSearchResponses.scala:704]: Sure, but the value of `feature.id` is surprising:

> omitting feature /eodata/Sentinel-2/MSI/L2A/2023/09/24/S2B_MSIL2A_20230924T103659_N0509_R008_T31UFS_20230924T132849.SAFE with tileId 31UFS

Comment (EmileSonneveld) [reply src/main/scala/org/openeo/opensearch/OpenSearchResponses.scala]: A bit long id indeed. But might be handy in the future

**Pull request merged**